test/Microsoft.AspNetCore.Routing.Tests/DispatcherApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder.Internal;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public class DispatcherApplicationBuilderExtensions
+    {
+        [Fact]
+        public void UseDispatcher_ServicesNotRegistered_Throws()
+        {
+            // Arrange
+            var app = new ApplicationBuilder(Mock.Of<IServiceProvider>());
+
+            // Act
+            var ex = Assert.Throws<InvalidOperationException>(() => app.UseDispatcher());
+
+            // Assert
+            Assert.Equal(
+                "Unable to find the required services. " +
+                "Please add all the required services by calling 'IServiceCollection.AddDispatcher' " +
+                "inside the call to 'ConfigureServices(...)' in the application startup code.",
+                ex.Message);
+        }
+
+        [Fact]
+        public void UseEndpoint_ServicesNotRegistered_Throws()
+        {
+            // Arrange
+            var app = new ApplicationBuilder(Mock.Of<IServiceProvider>());
+
+            // Act
+            var ex = Assert.Throws<InvalidOperationException>(() => app.UseEndpoint());
+
+            // Assert
+            Assert.Equal(
+                "Unable to find the required services. " +
```

Comment:
suggestion - `Unable to find the required Dispatcher services.`

Reply:
Hmm, the rest of the instructions say what you need to do, and it is a consistent message with when routing and MVC is missing.